packages/nextjs/hooks/scaffold-eth/useScaffoldWriteContract.ts

```diff
@@ -81,6 +83,16 @@ export function useScaffoldWriteContract<TContractName extends ContractName>(
     chainId: selectedNetwork.id as AllowedChainIds,
   });
 
+  const simulateAndNotifyError = async (params: WriteContractVariables<Abi, string, any[], Config, number>) => {
+    try {
+      await simulateContract(wagmiConfig, params);
+    } catch (error) {
+      const parsedError = getParsedError(error);
+      notification.error(parsedError);
+      throw error;
+    }
+  };
```

Comment:
We could have inlilned this in `sendContractWriteAsyncTx` function itself but notice the extra try catch here in this function. 

Couldn't inline it in `sendContractWriteAsyncTx` because if we show `notification.error` in `sendContractWriteAsyncTx` catch block then it might show error notification two times one when simulation failed other time incase the `useTransactor` fails. And even third time when user handles the error in his app.